sdk/tests/deqp/functional/gles3/es3fPrimitiveRestartTests.js

```diff
@@ -0,0 +1,724 @@
+        //gl.disable(gl.PRIMITIVE_RESTART_FIXED_INDEX);
+        //es3fPrimitiveRestartTests.GLU_CHECK_MSG('Disable primitive restart');
+
+        gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
+        es3fPrimitiveRestartTests.GLU_CHECK_MSG('Clear in renderWithoutRestart()');
+
+        // Draw, emulating primitive restart.
+
+        /** @type {number} */ var numIndices = this.getNumIndices();
+
+        DE_ASSERT(numIndices >= 0);
+
+        /** @type {number} */ var indexArrayStartNdx = 0; // Keep track of the draw start index - first index after a primitive restart, or initially the first index altogether.
+
+        for (var indexArrayNdx = 0; indexArrayNdx <= numIndices; indexArrayNdx++) { // \note Goes one "too far" in order to detect end of array as well.
+            if (indexArrayNdx >= numIndices || this.getIndex(indexArrayNdx) == restartIndex) {// \note Handle end of array the same way as a restart index encounter.
+                if (indexArrayStartNdx < numIndices) {
+                    // Draw from index indexArrayStartNdx to index indexArrayNdx-1 .
+
+                    this.draw(indexArrayStartNdx, indexArrayNdx - indexArrayStartNdx);
+                    es3fPrimitiveRestartTests.GLU_CHECK_MSG('Draw in renderWithoutRestart()');
+                }
+
+                indexArrayStartNdx = indexArrayNdx + 1; // Next draw starts just after this restart index.
+            }
+        }
+
+        es3fPrimitiveRestartTests.GLU_CHECK_MSG('renderWithoutRestart() end');
+    };
+
+    /**
+    * @param {number} index
+    */
+    es3fPrimitiveRestartTests.PrimitiveRestartCase.prototype.addIndex = function(index) {
+        if (this.m_indexType == es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_BYTE) {
+            DE_ASSERT(deMath.deInRange32(index, 0, es3fPrimitiveRestartTests.MAX_UNSIGNED_BYTE));
+            this.m_indicesUB.push(index); // deUint8
+        } else if (this.m_indexType == es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_SHORT) {
+            DE_ASSERT(deMath.deInRange32(index, 0, es3fPrimitiveRestartTests.MAX_UNSIGNED_SHORT));
+            this.m_indicesUS.push(index); // deUint16
+        } else if (this.m_indexType == es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_INT) {
+            DE_ASSERT(deMath.deInRange32(index, 0, es3fPrimitiveRestartTests.MAX_UNSIGNED_INT));
+            this.m_indicesUI.push(index); // // deUint32
+        } else
+            DE_ASSERT(false);
+    };
+
+    /**
+    * @param {number} indexNdx
+    * @return {number}
+    */
+    es3fPrimitiveRestartTests.PrimitiveRestartCase.prototype.getIndex = function(indexNdx) {
+        switch (this.m_indexType) {
+            case es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_BYTE:
+                return this.m_indicesUB[indexNdx]; //deUint32
+            case es3fPrimitiveRestartTests.IndexType.INDEX_UNSIGNED_SHORT:
+                return this.m_indicesUS[indexNdx]; //deUint32
```

Comment:
Starting from the refactoring of the code to use the closure library, we've been trying to remove those uint32/uint16 references, only leaving comments in some cases, while using plain number types.